cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/resource/EndpointOperations.java

```diff
@@ -15,13 +15,13 @@
 /**
  * Operations for {@code Endpoint}s.
  */
-public class EndpointOperations extends AbstractOperations<KubernetesClient, Endpoints, EndpointsList, DoneableEndpoints, Resource<Endpoints, DoneableEndpoints>> {
+class EndpointOperations extends AbstractReadyOperations<KubernetesClient, Endpoints, EndpointsList, DoneableEndpoints, Resource<Endpoints, DoneableEndpoints>, Void> {
```

Comment:
It's not clear to me why this class and the `PodOperations` one are the only operations classes to be package visible and not public.

Reply:
They're not really "first class" resources, but are "owned" by StatefulSets/Deployments and Services, respectively. We don't ever need to create or patch a bare Pod or Endpoint. Given that, there's no need to expose such functionality outside the package: We can let the corresponding owning operators take responsibility for using the PodOperations etc.